parser/src/ast/random_values.rs

```diff
@@ -0,0 +1,82 @@
+use super::Identifier;
+
+// RANDOM VALUES
+// ================================================================================================
+
+/// Declaration of random values for an AIR. Random values colud be represented by a named
+/// identifier which is used to identify a fixed size array of length `size` and empty `values`
+/// vector or by named identifier which is used to identify a `values` [RandomValue] vector and
+/// zero `size` field.
+///
+/// # Examples
+///
+/// If random values are declared in form
+///
+/// ```airscript
+/// random_values:
+///     rand: [15]
+///
+/// ```
+///
+/// created [RandomValues] instance will look like
+///
+/// `RandomValues{name: "rand", size: 15, values: []}`
+///
+/// If random values declared in form
+///
+/// ```airscript
+/// random_values:
+///     rand: [a, b[12]]
+///
+/// ```
+///
+/// created [RandomValues] instance will look like
+///
+/// `RandomValues{name: "rand", size: 0, values: [RandomValue{name: "a", size: 1}, RandomValue{name: "b", size: 12}]}`
+#[derive(Debug, Eq, PartialEq)]
+pub struct RandomValues {
+    name: Identifier,
+    size: usize,
+    values: Vec<RandomValue>,
```

Comment:
Using `bindings` might be better because technically we just bind name stored here to some random value, right? 

For now `name` is not using anywhere, I added it so that in the future it would be easier to implement a random value call by this name. So we allow users to create random values arrays with custom name, but `$` access could be done only with `$rand` name (so this feature is not actually working for now). As I mentioned in PR description 
> For now it is possible to create random values array with custom name, but accessing its values could be done only with $rand call.

I thought about it, my idea was to make `size = 0` as some kind of flag, showing that we are working with named random values. But otherwise this `values` array will be empty, so, indeed, I can simplify inserting random values into the symbol table by always storing size. I'll improve this part, thank you!

Reply:
I have a question related to this discussion: how would one access the random value `b[3]` if we have the following declaration `rand: [a, b[12], c]`?

Reply:
Great question. I think I have confused myself and made a mistake. In this "named" style of declaration I assume that the length of random values array is equal to the number of named declarations (e.g. `[a, b[12], c]` has length 3), and then I was working with elements as integers or arrays. But maybe it's incorrect and I should work with flattened array (e.g. `[a, b[12], c]` should have length 14). It will be better in all aspects.

Answering your question: for now it's possible only with `b[3]` named access, because current implementation only supports indexes to the each named variable (so `a` is `$rand[0]`, `b` is `$rand[1]` and `c` is `$rand[2]`), and that is my fault, I didn't understand well what I had to implement. I'll fix it now

Reply:
> Answering your question: for now it's possible only with b[3] named access, because current implementation only supports indexes to the each named variable (so a is $rand[0], b is $rand[1] and c is $rand[2]), and that is my fault, I didn't understand well what I had to implement. I'll fix it now

Yes, I think this is not the correct approach. When accessing these using `$rand` we should flatten the array as you've mentioned

Reply:
Thank you guys, it makes more sense to me now.